src/js/player.js

```diff
@@ -1747,10 +1747,11 @@ class Player extends Component {
    */
   duration(seconds) {
     if (seconds === undefined) {
-      return this.cache_.duration || 0;
+      // return NaN if the duration is not known
+      return this.cache_.duration !== undefined ? this.cache_.duration : NaN;
     }
 
-    seconds = parseFloat(seconds) || 0;
+    seconds = parseFloat(seconds);
```

Comment:
Ah, I see. 🤔
I guess maybe this is fine since `parseFloat()` will return `NaN` for bad values. So, we'd just want to make sure the rest of this method handles `NaN` correctly, if it doesn't already.